crownstone_ble/core/CrownstoneBle.py

```diff
@@ -108,22 +109,105 @@ async def getCrownstonesByScanning(self, scanDuration=3):
+            raise CrownstoneBleException(BleError.NO_SCANS_RECEIVED, f'During the {scanDuration} seconds of scanning, no advertisement was received from this address.')
+
+        return result
+
+
+
+    async def getMode(self, address, scanDuration=3) -> CrownstoneOperationMode:
+        """
+        This will wait until it has received an advertisement from the Crownstone with the specified address. Once it has received an advertisement, it knows the mode.
+
+        We will return once we know.
+
+        It can throw the following CrownstoneBleException
+        - BleError.NO_SCANS_RECEIVED
+            We have not received any scans from this Crownstone, and can't say anything about it's state.
+        """
+        _LOGGER.debug(f"getMode address={address} scanDuration={scanDuration}")
+        checker = ModeChecker(address, None)
+        subscriptionId = BleEventBus.subscribe(BleTopics.rawAdvertisement, lambda scanData: checker.handleAdvertisement(scanData))
+        await self.ble.scan(duration=scanDuration)
+        BleEventBus.unsubscribe(subscriptionId)
+        result = checker.getResult()
+
+        if result is None:
+            raise CrownstoneBleException(BleError.NO_SCANS_RECEIVED, f'During the {scanDuration} seconds of scanning, no advertisement was received from this address.')
+
+        return result
+
+
+    async def waitForMode(self, address, requiredMode: CrownstoneOperationMode, scanDuration=5):
+        """
+        This will wait until it has received an advertisement from the Crownstone with the specified address. Once it has received an advertisement, it knows the mode. We will
+        scan for the scanDuration amount of seconds or until the Crownstone is in the required mode.
+
+        It can throw the following CrownstoneBleException
+        - BleError.NO_SCANS_RECEIVED
+            We have not received any scans from this Crownstone, and can't say anything about it's state.
+        - BleError.DIFFERENT_MODE_THAN_REQUIRED
+            During the {scanDuration} seconds of scanning, the Crownstone was not in the required mode.
+        """
+        _LOGGER.debug(f"waitForMode address={address} requiredMode={requiredMode} scanDuration={scanDuration}")
+        checker = ModeChecker(address, requiredMode, True)
+        subscriptionId = BleEventBus.subscribe(BleTopics.rawAdvertisement, lambda scanData: checker.handleAdvertisement(scanData))
+        await self.ble.scan(duration=scanDuration)
+        BleEventBus.unsubscribe(subscriptionId)
+        result = checker.getResult()
+
+        if result is None:
+            raise CrownstoneBleException(BleError.NO_SCANS_RECEIVED, f'During the {scanDuration} seconds of scanning, no advertisement was received from this address.')
+        if result != requiredMode:
+            raise CrownstoneBleException(BleError.DIFFERENT_MODE_THAN_REQUIRED, f'During the {scanDuration} seconds of scanning, the Crownstone was not in the required mode..')
+
+        return result
```

Comment:
```suggestion
        return
```

Reply:
Actually, this return is not required at all.